auth/src/main/java/com/firebase/ui/auth/ui/accountmanagement/EditDisplayNameActivity.java

```diff
@@ -0,0 +1,86 @@
+package com.firebase.ui.auth.ui.accountmanagement;
+
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.bumptech.glide.Glide;
+import com.firebase.ui.auth.R;
+import com.firebase.ui.auth.ui.BaseHelper;
+import com.firebase.ui.auth.ui.FlowParameters;
+import com.firebase.ui.auth.ui.TaskFailureLogger;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.firebase.auth.UserProfileChangeRequest;
+
+public class EditDisplayNameActivity extends SaveFieldActivity {
+    private static final String TAG = "EditDisplayNameAct";
+    private EditText mDisplayName;
+
+    public static Intent createIntent(Context context, FlowParameters flowParameters) {
+        return BaseHelper.createBaseIntent(context, EditDisplayNameActivity.class, flowParameters);
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_edit_display_name);
+        mDisplayName = (EditText) findViewById(R.id.display_name);
+        mDisplayName.setText(mActivityHelper.getCurrentUser().getDisplayName());
+
+        displayProfilePicture();
+
+        TextView removeLink = (TextView) findViewById(R.id.remove_profile_image_link);
+        removeLink.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                removeProfilePicture();
+            }
+        });
+    }
+
+    @Override
+    protected void onSaveMenuItem() {
+        mActivityHelper.getCurrentUser().updateProfile(
+                new UserProfileChangeRequest.Builder()
+                        .setDisplayName(mDisplayName.getText().toString())
+                        .build())
+                .addOnSuccessListener(new OnSuccessListener<Void>() {
+                    @Override
+                    public void onSuccess(Void aVoid) {
+                        finish();
+                    }
+                })
+                .addOnFailureListener(new TaskFailureLogger(TAG, "Failed to save display name"));
+    }
+
+    private void displayProfilePicture() {
```

Comment:
Slightly confused: why is the option to remove the profile picture part of the edit name activity? Is this just because the edit button is for the whole row?